contracts/utils/deploy.js

```diff
@@ -108,24 +111,30 @@ const impersonateGuardian = async () => {
  *
  * @param {Array<Object>} proposalArgs
  * @param {string} description
- * @param {boolean} whether to use the V1 governor (e.g. MinuteTimelock)
+ * @param {opts} Options
+ *   v1: whether to use the V1 governor (e.g. MinuteTimelock)
+ *   governorAddr: address of the governor contract to send the proposal to
+ *   guardianAddr: address of the guardian (aka the governor's admin) to use for sending the queue and execute tx
  * @returns {Promise<void>}
  */
-const executeProposal = async (proposalArgs, description, v1 = false) => {
+const executeProposal = async (proposalArgs, description, opts = {}) => {
   if (isMainnet || isRinkeby) {
     throw new Error("executeProposal only works on local test network");
   }
 
-  const { deployerAddr, guardianAddr } = await hre.getNamedAccounts();
+  const namedAccounts = await hre.getNamedAccounts();
+  const deployerAddr = namedAccounts.deployerAddr;
+  const guardianAddr = opts.guardianAddr || namedAccounts.guardianAddr;
+
   const sGuardian = hre.ethers.provider.getSigner(guardianAddr);
   const sDeployer = hre.ethers.provider.getSigner(deployerAddr);
 
   if (isFork) {
-    await impersonateGuardian();
+    await impersonateGuardian(opts.guardianAddr);
   }
 
   let governorContract;
-  if (v1) {
+  if (opts.v1) {
```

Comment:
Do we need to keep this v1 stuff?

Reply:
yeah you are right, not necessary. let me clean it up...